packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -97,7 +97,9 @@ export const KeyboardNavigationMixin = (superClass) =>
 
     /** @private */
     get __rowFocusMode() {
-      return this.__isRow(this._itemsFocusable);
+      return (
+        this.__isRow(this._itemsFocusable) || this.__isRow(this._headerFocusable) || this.__isRow(this._footerFocusable)
+      );
```

Comment:
Change addressed by ["should return to cell focus mode" test](https://github.com/vaadin/web-components/pull/2998/files#diff-318f8ea00fbad8b55d74575cf034e581cbfb4798b21f57dc73bd207b2e8db456R533)